File: ProtocolDemo/ViewController.m
//
//  ViewController.m
//  ProtocolDemo
//
//  Created by BooB on 15/8/8.
//  Copyright (c) 2015年 BooB. All rights reserved.
//

#import "ViewController.h"
#import "CELL_STRUCT_Common.h"
#import "CELL_STRUCT.h"

@interface ViewController ()
AS_CELL_STRUCT_JXLCOMMON(cell00)
AS_CELL_STRUCT_JXLCOMMON(cell01)
@end

@implementation ViewController
GET_CELL_STRUCT_WITH(cell00, cell00)
GET_CELL_STRUCT_WITH(cell01, cell01)


GET_CELL_SELECT_ACTION(cellstruct)
{
    NSLog(@"%@ 点击",cellstruct.title);
}
- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.dataDictionary setObject:self.cell_struct_cell00 forKey:KEY_INDEXPATH(0, 0)];
    [self.dataDictionary setObject:self.cell_struct_cell01 forKey:KEY_INDEXPATH(0, 1)];
    
    for (int INDEX = 0; INDEX < 50; INDEX ++) {
        CELL_STRUCT * cellstruct = [CELL_STRUCT_Common cell_x_x_struct:[NSString stringWithFormat:@"row %d",INDEX] detailvalue:nil target:self selectAction:DEFAULT_CELL_SELECT_ACTION];
        [self.dataDictionary setObject:cellstruct forKey:KEY_INDEXPATH(1, INDEX)];
    }
    
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: demo/TestSwiftHBKit/TestSwiftHBKit/ViewController.swift
//
//  ViewController.swift
//  TestSwiftHBKit
//
//  Created by boob on 16/12/2.
//  Copyright © 2016年 YY.COM. All rights reserved.
//

import UIKit

class ViewController: HBBaseTableViewController {
  
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        
//        for index in 1...100{
//            let cs = HBCellStruct.cell_x_x_struct("cell" + String(index), detailvalue: "", target: self, selectAction: Selector(("selectAction:"))) as HBCellStruct
//            cs.dictionary.setObject("random", forKey: key_cellstruct_background as NSCopying);
//            self.dataDictionary.setObject(cs, forKey: self.key_indexpath(0, row: index) as NSCopying)
//        }
        
        self.loadjsonfileConfig("TestJson", watch_directory: "resource")
//        self.loadplistConfig("Testplistvc", watch_directory: "resource")
        
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func selectAction(cs:HBCellStruct)->(){
        print(cs.description);
    }

}

